managed/services/management/backup/artifacts_service.go

```diff
@@ -133,7 +133,11 @@ func (s *ArtifactsService) ListPitrTimeranges(
 	}
 
 	if artifact.Mode != models.PITR {
-		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact")
+		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact.")
+	}
+
+	if artifact.IsShardedCluster {
+		return nil, status.Errorf(codes.FailedPrecondition, "Getting PITR timeranges is not supported for sharded cluster artifacts.")
```

Comment:
But why?

Reply:
The algorithm that parsing oplog files doesn't split the files by shards, therefore prvides wrong data for sharded setups. You can remember, I mentioned that earlier.